src/graph/Graph.cpp

```diff
@@ -64,8 +64,11 @@ void Graph::run()
 
 	RGL_DEBUG("Running graph with {} nodes", nodesInExecOrder.size());
 
-	RaytraceNode::Ptr rt = Node::getExactlyOne<RaytraceNode>(nodesInExecOrder);
-	rt->setFields(fieldsToCompute);
+	// If Graph has RaytraceNode set fields to compute
+	if (!Node::filter<RaytraceNode>(nodesInExecOrder).empty()) {
+		RaytraceNode::Ptr rt = Node::getExactlyOne<RaytraceNode>(nodesInExecOrder);
```

Comment:
To be discussed, what about many raytrace nodes in a single graph.

Reply:
Great catch. I'd apply this change in another PR.

Reply:
IMHO having a single RaytraceNode is a reasonable assumption allowing for some simplifications.